Service/ListingAdDataCreator/ListingAdDataCreatorService.php

```diff
@@ -102,19 +105,27 @@ private function getFilteredProductsQuery(Application $app, Request $request)
      *
      * @param Application $app
      * @param Request $request
+     * @param EntityManager $em
      * @return \Doctrine\ORM\QueryBuilder
      */
-    private function getProductQueryBuilder(Application $app, Request $request)
+    private function getProductQueryBuilder(Application $app, Request $request, $em)
     {
         $session = $request->getSession();
         $viewData = $session->get('eccube.admin.product.search', array());
-        $searchForm = $app['form.factory']->create('admin_search_product');
-        $searchData = FormUtil::submitAndGetData($searchForm, $viewData);
-        if (isset($viewData['link_status']) && strlen($viewData['link_status'])) {
-            $searchData['link_status'] = $app['eccube.repository.master.disp']->find($viewData['link_status']);
-        }
-        if (isset($viewData['stock_status'])) {
-            $searchData['stock_status'] = $viewData['stock_status'];
+        $searchData = $viewData;
```

Comment:
I think the source code will not be good, it will duplicate the code. Performance will be slow.